src/GafferOSL/OSLShader.cpp

```diff
@@ -1163,12 +1246,76 @@ Plug *loadShaderParameter( const OSLQuery &query, const OSLQuery::Parameter *par
 	return result;
 }
 
-void loadShaderParameters( const OSLQuery &query, Gaffer::Plug *parent, const CompoundData *metadata, const std::string &prefix )
+template <typename PlugType>
+Gaffer::Plug *loadSplineParameterFromDefault( const InternedString &name, const typename PlugType::ValueType &defaultValue, Gaffer::Plug *parent )
 {
+	PlugType *existingPlug = parent->getChild<PlugType>( name );
+	if( existingPlug && existingPlug->defaultValue() == defaultValue )
+	{
+		return existingPlug;
+	}
+
+	typename PlugType::Ptr plug = new PlugType( name, parent->direction(), defaultValue, Plug::Default );
 
+	// TODO - was it intentional that this logic was absent from previous spline processing?
+	if( existingPlug )
+	{
+		PlugAlgo::replacePlug( parent, plug );
+	}
+	else
+	{
+		parent->setChild( name, plug );
+	}
+
+	return plug.get();
+}
+
+void loadShaderParameters( const std::string &shaderName, const OSLQuery &query, Gaffer::Plug *parent, const CompoundData *metadata, const std::string &prefix )
+{
 	// Make sure we have a plug to represent each parameter, reusing plugs wherever possible.
 
 	set<const Plug *> validPlugs;
+
+	// Spline parameters are a nasty special case because multiple shader
+	// parameters become a single plug on the node, so we deal with them
+	// first, and use this set to record parameters that have already been
+	// processed as part of a spline.
+
+	std::set<std::string> parametersAlreadyProcessed;
+	std::map<std::string, SplineParameters > splineParameters;
+
+	for( size_t i = 0; i < query.nparams(); ++i )
+	{
+		// We will check whether this parameter fits as the positions parameter of a spline
+		const OSLQuery::Parameter *positionsParameter = query.getparam( i );
+		const Plug::Direction direction = positionsParameter->isoutput ? Plug::Out : Plug::In;
+		if( direction != parent->direction() )
+		{
+			continue;
+		}
+
+		if( !boost::starts_with( positionsParameter->name.string(), prefix ) )
+		{
+			continue;
+		}
+
+		const string name = positionsParameter->name.string().substr( prefix.size() );
+		if( name.find( "." ) != string::npos )
+		{
+			// Member of a struct - will be loaded when the struct is loaded
+			continue;
+		}
+
+		SplineParameters splineP = splineParametersFromPositions( shaderName, query, positionsParameter, prefix, parametersAlreadyProcessed );
+
+		// This is a very weird way of saying "The variant isn't empty", but it seems to be the modern C++ convention )
```

Comment:
Perhaps this might be clearer if `splineDefault` only had the two _valid_ types, and `splineParametersFromPositions()` returned `std::optional<SplineParameters>`? It's more obvious that a value isn't useable if you can't even _get_ the value, rather than getting it and checking that its fields seem to have what you want.

Reply:
The "very weird" comment is still there.